apps/api/src/app/events/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -175,4 +201,52 @@ export class TriggerEvent {
       );
     }
   }
+
+  private async getProviderIdsForTemplate(
+    userId: string,
+    organizationId: string,
+    environmentId: string,
+    template: NotificationTemplateEntity
+  ): Promise<Map<ChannelTypeEnum, string>> {
+    const providers = new Map<ChannelTypeEnum, string>();
+
+    for (const step of template?.steps) {
+      const type = step.template?.type;
+      if (type) {
+        const channelType = STEP_TYPE_TO_CHANNEL_TYPE.get(type);
+
+        if (channelType) {
+          const provider = await this.getProviderId(userId, organizationId, environmentId, channelType);
+          if (provider) {
+            providers.set(channelType, provider);
+          } else {
+            providers.set(channelType, InAppProviderIdEnum.Novu);
+          }
+        }
+      }
+    }
+
+    return providers;
+  }
+
+  private async getProviderId(
+    userId: string,
+    organizationId: string,
+    environmentId: string,
+    channelType: ChannelTypeEnum
+  ): Promise<string> {
+    const integrations = await this.getDecryptedIntegrations.execute(
+      GetDecryptedIntegrationsCommand.create({
```

Comment:
I think we can pass here findOne:true as a param so on the command, that way you wont fetch by 'find'

Reply:
Checking `GetDecryptedIntegrations` code I do not find a big difference between both options as downstream still operates as an array and does all the checks adding the one entity found to an array.
https://github.com/novuhq/novu/blob/e32b5c2bb4d48570342769a9305fbe1e2d2a1209/apps/api/src/app/integrations/usecases/get-decrypted-integrations/get-decrypted-integrations.usecase.ts#L30

Reply:
You are right, the only difference as far as I know will be the query operation find/findOne when 'find' will query all the collection, and 'findOne' will stop after the first find.